base/inference.jl

```diff
@@ -1403,8 +1403,8 @@ function smerge(sa::Union{NotFound,VarState}, sb::Union{NotFound,VarState})
     VarState(tmerge(sa.typ, sb.typ), sa.undef | sb.undef)
 end
 
-tchanged(n::ANY, o::ANY) = is(o,NF) || (!is(n,NF) && !(n ⊑ o))
-schanged(n::ANY, o::ANY) = is(o,NF) || (!is(n,NF) && !issubstate(n, o))
+tchanged(n::ANY, o::ANY) = o===NF || (n!==NF && !(n ⊑ o))
+schanged(n::ANY, o::ANY) = o===NF || (n!==NF && !issubstate(n, o))
```

Comment:
Inference performance regressions probably weren't caught by nanosoldier. I'd be surprised if this didn't regress, but it's probably not a big deal.